src/middleware.ts

```
@@ -83,13 +83,18 @@ function hasPermission(role: string, pathname: string): boolean {
 
-function getDefaultPath(role: string): string {
+function getDefaultPath(role: string, request: NextRequest): string {
   switch (role) {
     case 'ROLE_GUEST':
       return '/survey';
     case 'ROLE_ADMIN':
       return '/admin';
     case 'ROLE_USER':
+      const isNewUserCookie = request.cookies.get('isNewUser')?.value;
+      if (isNewUserCookie === 'true') {
+        return '/onboarding';
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)